logstash-core/lib/logstash/agent_plugin_manager.rb

```diff
@@ -0,0 +1,65 @@
+# encoding: utf-8
+require "logstash/agent"
+
+module LogStash module AgentPluginManager
+  DEFAULT_AGENT_NAME = :default
+  class DuplicatePluginNameError < Error; end
+
+  REGISTRY = {}
+
+  # Reset plugin registry to just the default plugin
+  def self.reset!
+    REGISTRY.clear
+    REGISTRY[DEFAULT_AGENT_NAME] = LogStash::Agent
+  end
+
+  reset!
```

Comment:
That needs a comment explaining that the purpose is to set the default agent when the class is loaded. That DRYs up the code or I'd have to put it in the `REGISTRY` declaration above.